ZachG update parametric documentation and add contributing doc (#3325)
DataDog/system-tests: docs/scenarios/parametric.md

```

- This test case runs against all the APM libraries and is parameterized with two different environments specifying two different values of the environment variable `DD_ENV`.
- `test_library.dd_start_span` creates a new span using the shared HTTP interface.
- The request is sent to a HTTP server by language. Implementations can be found in `utils/build/docker/<lang>/parametric`. More information in [Http Server Implementations](#http-server-implementations).
- Data is flushed to the test agent after the with test_library block closes.

> - Traces are retrieved using the `test_agent` fixture and we assert that they look the way we'd expect.

Author: ```suggestion
- Data (usually traces) are retrieved using the `test_agent` fixture and we assert that they look the way we'd expect.
```

> - Traces are retrieved using the `test_agent` fixture and we assert that they look the way we'd expect.

Author: I think it can do other data, and if it can't it may someday be able to, so I'll go with data and clarify usually traces

```

- This test case runs against all the APM libraries and is parameterized with two different environments specifying two different values of the environment variable `DD_ENV`.
- The test case creates a new span and sets a tag on it using the shared HTTP interface.

> - The implementations of the HTTP interface, by language, are in `utils/build/docker/<lang>/parametric`. See here for exact locations per langugage: [Http Server Implementations](#http-server-implementations) section for more details.

Author: ```suggestion
- The implementations of the HTTP interface, by language, are in `utils/build/docker/<lang>/parametric`. See here for exact locations per langugage: [Http Server Implementations](#http-server-implementations) for more details.
```

- Data is flushed to the test agent after the with test_library block closes.

> - Data is retrieved using the `test_agent` fixture and asserted on.

Author: ```suggestion
- Traces are retrieved using the `test_agent` fixture and we assert that they look the way we'd expect.
```

### Shared Interface

To view the available HTTP endpoints , follow these steps:
Note: These are based off of the Python tracer's http server which should be held as the standard example interface across implementations.

1. `./utils/scripts/parametric/run_reference_http.sh`
2. Navigate to http://localhost:8000/docs in your web browser to access the documentation.
3. You can download the OpenAPI schema from http://localhost:8000/openapi.json. This schema can be imported into tools like [Postman](https://learning.postman.com/docs/integrations/available-integrations/working-with-openAPI/) or other API clients to facilitate development and testing.

> 

Author: ```suggestion
Not all endpoint implementations per language are up to spec with regards to their parameters and return values. To see these please reference the [feature parity board](https://feature-parity.us1.prod.dog/#/?runDateFilter=7d&feature=339)

```

This architecture allows us to ensure that all tracers conform to the same interface and behavior, making it easier to maintain consistency across different languages and implementations.

#### Http Server Implementations

> The http server implementations for each tracer can be found at the following locations:

Author: Sure!

This architecture allows us to ensure that all tracers conform to the same interface and behavior, making it easier to maintain consistency across different languages and implementations.

#### Http Server Implementations

The http server implementations for each tracer can be found at the following locations:

> [Python](/utils/build/docker/python/parametric/apm_test_client/server.py)
> 
> [Ruby](utils/build/docker/ruby/parametric/server.rb)
> 
> [Php](utils/build/docker/php/parametric/server.php)
> 
> [Nodejs](utils/build/docker/nodejs/parametric/server.js)
> 
> [Java Datadog](utils/build/docker/java/parametric/src/main/java/com/datadoghq/trace/opentracing/controller/OpenTracingController.java)
> 
> [Java Otel](utils/build/docker/java/parametric/src/main/java/com/datadoghq/trace/opentelemetry/controller/OpenTelemetryController.java)
> 
> [Dotnet Datadog](utils/build/docker/dotnet/parametric/Endpoints/ApmTestApi.cs)
> 
> [Dotnet Otel](utils/build/docker/dotnet/parametric/Endpoints/ApmTestApiOtel.cs)
> 
> [Go Datadog](utils/build/docker/golang/parametric/main.go)
> 
> [Go Otel](utils/build/docker/golang/parametric/otel.go)

Reviewer: Markdown list

```suggestion
* [Python](/utils/build/docker/python/parametric/apm_test_client/server.py)
* [Ruby](utils/build/docker/ruby/parametric/server.rb)
* [Php](utils/build/docker/php/parametric/server.php)
* [Nodejs](utils/build/docker/nodejs/parametric/server.js)
* [Java Datadog](utils/build/docker/java/parametric/src/main/java/com/datadoghq/trace/opentracing/controller/OpenTracingController.java)
* [Java Otel](utils/build/docker/java/parametric/src/main/java/com/datadoghq/trace/opentelemetry/controller/OpenTelemetryController.java)
* [Dotnet Datadog](utils/build/docker/dotnet/parametric/Endpoints/ApmTestApi.cs)
* [Dotnet Otel](utils/build/docker/dotnet/parametric/Endpoints/ApmTestApiOtel.cs)
* [Go Datadog](utils/build/docker/golang/parametric/main.go)
* [Go Otel](utils/build/docker/golang/parametric/otel.go)
```

This architecture allows us to ensure that all tracers conform to the same interface and behavior, making it easier to maintain consistency across different languages and implementations.

#### Http Server Implementations

The http server implementations for each tracer can be found at the following locations:

> 

Author: ```suggestion
*Note:* For some languages there is both an Otel and a Datadog server. This is simply to separate the available Otel endpoints from the available Datadog endpoints that can be hit by the client. If a language only has a single server, then both endpoints for Otel and Datadog exist there.

```
